lektor/cli.py

```diff
@@ -265,6 +265,14 @@ def deploy_cmd(ctx, server, output_path, **credentials):
     env = ctx.get_env()
     config = env.load_config()
 
+    if (server is not None and
+            (len(server) == 2 and server[0] == '-') or
+            (len(server) > 2 and server[:2] == '--')):
+
+        # We captured a publisher-specific option as the "server" arg.
+        click.get_current_context().args.insert(0, server)
+        server = None
```

Comment:
Is there some way to configure click.Command to avoid this? If I allow extra args and do "lektor deploy --draft", then "--draft" is parsed as the "server" argument. Of course this isn't a problem if the server name is specified: "lektor deploy servername --draft".